### ISSUE TITLE
It is nice you included alternativees for this Atom package. But also inlcude reasons to choose this particular package for linting python in Atom.

### ISSUE THREAD

**Issue by twisp007** [elarivie/atom-python-linters#5]
I see you have included a list of alternative packages for python but it doesn't really help why one should your package.

Please include in the README:
What are the features of your package that stand out. Or even your motivation for creating a new package while the ones you listed already exist.

**Comment by elarivie** [comment]
Hello @twisp007,

You have unusual issue questions that you are asking 😀

But I'm glad to answer them.


```I see you have included a list of alternative packages for python```

Because I am not a sales person 😀 Pick the setup that you want for your own development environment/preference.

**Selling speech:** I personnally use python-linters package to lint python code within atom since it handles every lint tool, nothing to configure, only one package to install in atom, it just works right away with the setup.cfg file present in your projects.

I am not a fan of vendor lock-in concept, it is nice to have alternatives, if you don't like python-linters, if you don't want to use setup.cfg... there are alternatives and they are provided for you in the readme.

```it doesn't really help why one should use your package```

Well, I am not in competition with those other atom packages.  Python-linters embed all their lint capacity in one single atom package.  In fact if you want to use one or many of them you can use them in parallel with python-linters (you can deactivate individual lint tool within atom python-linters settings to avoid double processing)

```your motivation for creating a new package while the ones you listed already exist.```


**TLDR:** Because I personnally need/use it

## Timeline


First you may have noticed that I am also the author of [linter-mypy](https://atom.io/packages/linter-mypy) 
which is the first atom package I wrote because at that time, there was no Mypy integration within atom and I wanted one so I did develop it.  It was my first atom package so I did some design errors…

**Design Error 1**- I implemented linter-mypy with Coffeescript because the atom package template that I found was done in that language and thought that atom package have to be done in that language (I learned Coffeescript and came to the conclusion that Coffeescript is actually worst than Javascript and later discovered that you can implement atom package in other language 😀)

**Design Error 2**- I did implement within the atom settings for linter-mypy all Mypy options available (currently ~35 settings) which is a pain to maintain.  I have a dedicated unit-test to detect Mypy CLI interface changes which occurs every few weeks.

**Design Error 3**- Having lint setting within the IDE is not a great idea because then if you want to lint your project using the CLI (For linting within CI for example)... you can't reuse your IDE settings, so you end-up with two somewhat similar configuration at two different place (not really DRY).

**Design Error 4**- I don't want to configure atom package**s** settings whenever I reinstall atom.

**Design Error 5**- Having lint settings within the IDE is also not useful if you have more than one project with different settings.

Back then I was using all of those lint package:
    https://atom.io/packages/linter-flake8
    https://atom.io/packages/linter-mypy
    https://atom.io/packages/linter-pydocstyle
    https://atom.io/packages/linter-pylint

Then one of the above lint package kept crashing and was generating annoying pop-up issue and it was barely possible to do anything so I had to deactive it but still wanted its lints message…

* Using the knowledge I acquired by implementing linter-mypy
* Using this linter package [undocumented feature](https://github.com/steelbrain/linter/issues/1670)
* Using the fact that [all of the lint tools support setup.cfg configuration file](http://renesd.blogspot.com/2017/02/setupcfg-solution-to-python-config-file.html).

I therefore had the idea of having ONE atom package which embed them all and just works by default without having to touch any settings seemed a good idea to implement.

I decided to implement a KISS package `python-linters` to solve the problems I had.  My plan was to implement python-linters without the above design errors and then delete linter-mypy.

**Note:** I created python-linters but I didn't delete linter-mypy and keep maintaining it because there are ~15k download and I don't want to mess with those people setup.

```Please include in the README What are the features of your package that stand out```

Effectively it could help people to make tools decision, I'll look into it in the next few days.

Hope it answers your questions.

**Comment by twisp007** [comment]
This gave me a much clearer perspective. Thank you!
